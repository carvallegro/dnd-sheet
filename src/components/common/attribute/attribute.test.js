import React from 'react'
import { shallow } from 'enzyme'

import Attribute from './index'

describe('<Attribute />', () => {
  it('renders', () => {
    const wrapper = shallow(<Attribute />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with values', () => {
    const wrapper = shallow(<Attribute value={10} modifier={2} name="Test" />)
    expect(wrapper).toMatchSnapshot()
  })

  it('renders with negative modifier', () => {
    const wrapper = shallow(<Attribute modifier={-2} />)
    expect(wrapper).toMatchSnapshot()
  })
})
