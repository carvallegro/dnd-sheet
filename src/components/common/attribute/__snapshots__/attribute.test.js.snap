// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Attribute /> renders 1`] = `
<styled.div>
  <styled.p />
  <_default />
  <styled.div>
    <_default />
  </styled.div>
</styled.div>
`;

exports[`<Attribute /> renders with negative modifier 1`] = `
<styled.div>
  <styled.p />
  <_default
    value={-2}
  />
  <styled.div>
    <_default />
  </styled.div>
</styled.div>
`;

exports[`<Attribute /> renders with values 1`] = `
<styled.div>
  <styled.p>
    Test
  </styled.p>
  <_default
    value={2}
  />
  <styled.div>
    <_default
      value={10}
    />
  </styled.div>
</styled.div>
`;
